src/Models/BaseElement.php

```diff
@@ -656,15 +656,9 @@ public function getRenderTemplates($suffix = '')
      */
     public function updateFromFormData($data)
     {
-        $cmsFields = $this->getCMSFields();
-
-        foreach ($data as $field => $datum) {
-            $field = $cmsFields->dataFieldByName($field);
-
-            if (!$field) {
-                continue;
-            }
-
+        $cmsFields = $this->getCMSFields()->saveableFields();
```

Comment:
Getting the `saveableFields()` explicitly - things like `LiteralField` don't need to be touched here.